### DISCUSSION TITLE
Shared UI Do I have to use typescript

### DISCUSSION THREAD

**Discussion by Jamiecharris** [vercel/turborepo#2579] [Help] [answered]
Hi, I have moved my nextjs apps over and move some of the UI components in the shared UI part of turbo repo.

However I'm getting errors in the shared ui. My ui files are .js I think turbo repo what's them as typescript? Basically can I use.js components in the shared ui library?

**Comment by nathanhammond** [comment, marked as answer]
You can do whatever you want, this is not related to anything other than your build configuration. Nothing to do with Turborepo.
